### PR TITLE
Fixes #1605 - Adds tests for tcms.core.middleware

### LINKED ISSUE #1605: Automated tests for missing coverage in tcms.core.middleware
Kiwi TCMS needs additional automated test(s) as mentioned in subject.

**Definition of Done**:
- full coverage for the module or class mentioned in the title. This means both positive/negative scenarios where this makes sense or scenarios which traverse different branches of the code where that makes sense.
- test class/test method names are descriptive and reflect the behavior of the subject under test.  A good scheme to follow is `Test<ClassUnderTest>.test_[who][what][expected_results]`
- CI pass
- All code review passed & PR merged

Hints:
- Coverage report is at https://codecov.io/gh/kiwitcms/Kiwi/tree/master/tcms
- Devel getting started docs: https://kiwitcms.readthedocs.io/en/latest/contribution.html#testing
- Django testing documentation: https://docs.djangoproject.com/en/3.0/topics/testing/

*This issue is part of Kiwi TCMS open source bounty program. For more information see the link(s) in [bounty-program milestone](https://github.com/kiwitcms/Kiwi/milestone/4)*

### PR DESCRIPTION
**Pull request by pandafy**
Closes #1605 

------

Initially, I chose to use actual GET/POST requests and test the working of the middlewares, but I changed my mind after looking into [Django's source code](https://github.com/django/django/blob/master/tests/middleware/tests.py).  I tried to follow how they have written tests for their middlewares.

### CODE REVIEW

File: tcms/core/tests/test_middleware.py [hunk 1]
```diff
@@ -16,3 +23,23 @@ def test_unapplied_migrations(self):
...
+
+class TestCsrfDisableMiddleware(unittest.TestCase):
+    def test_prcess_view(self):
+        request = HttpRequest()
+        middleware = CsrfDisableMiddleware()
+        middleware.process_view(request,
+                                _callback=None,
+                                _callback_args=None,
+                                _callback_kwargs=None)
+        self.assertEqual(request._dont_enforce_csrf_checks, True)
```
Comment (atodorov) [inline tcms/core/tests/test_middleware.py:36]: This middleware should be removed and thus doesn't need tests, see https://github.com/kiwitcms/Kiwi/issues/297

File: tcms/core/tests/test_middleware.py [hunk 2]
```diff
@@ -16,3 +23,23 @@ def test_unapplied_migrations(self):
...
+
+
+class TestCheckSettingsMiddleware(test.TransactionTestCase):
+    @mock.patch('django.contrib.messages.add_message')
+    def test_process_request(self, mocked_add_message):
+        request = HttpRequest()
+        middleware = CheckSettingsMiddleware()
+        middleware.process_request(request)
+        mocked_add_message.assert_called_once()
```
Comment (atodorov) [inline tcms/core/tests/test_middleware.py:45]: This test scenario is no good. It doesn't setup conditions in the SUT and doesn't assert any of the desired results other than the fact that the middleware was executed. This tells us nothing about whether or not this middleware executed successfully.  See the rest of this file for example.

**Review by atodorov — CHANGES_REQUESTED** [review]

File: tcms/core/tests/test_middleware.py (hunk 1 above)
Comment (pandafy) [reply tcms/core/tests/test_middleware.py]: I have removed the test. :smile:

File: tcms/core/tests/test_middleware.py (hunk 2 above)
Comment (pandafy) [reply tcms/core/tests/test_middleware.py]: I have updated this test according to `TestCheckUnappliedMigrationsMiddleware`

File: tcms/core/tests/test_middleware.py [hunk 3]
```diff
@@ -10,9 +11,19 @@
 class TestCheckUnappliedMigrationsMiddleware(test.TransactionTestCase):
     def test_unapplied_migrations(self):
         call_command('migrate', 'bugs', 'zero', verbosity=2, interactive=False)
-        unapplied_migration_message = 'unapplied migration(s). See '\
-            '<a href="https://kiwitcms.readthedocs.io/en/latest/'\
-            'installing_docker.html#initial-configuration-of-running-'\
+        unapplied_migration_message = 'unapplied migration(s). See ' \
+            '<a href="https://kiwitcms.readthedocs.io/en/latest/' \
+            'installing_docker.html#initial-configuration-of-running-' \
```
Comment (atodorov) [inline tcms/core/tests/test_middleware.py:16]: These changes are unrelated and cause a conflict. 

In addition we've started using black to format the code. All existing code has been reformatted and a CI check has been added. Rebase to latest master, resolve conflicts and update where necessary.

File: tcms/core/tests/test_middleware.py [hunk 4]
```diff
@@ -10,9 +11,19 @@
...
+
+
+class TestCheckSettingsMiddleware(test.TransactionTestCase):
+    def test_process_request(self):
+        base_url_message = 'Base URL is not configured! See ' \
+            '<a href="https://kiwitcms.readthedocs.io/en/latest/admin.html' \
+            '#configure-kiwi-s-base-url">documentation</a> and <a href="' \
+            '/admin/sites/site/1/change/">change it</a>!'
+        response = self.client.get('/', follow=True)
+        self.assertContains(response, base_url_message)
```
Comment (atodorov) [inline tcms/core/tests/test_middleware.py:29]: Missing test scenario where Base URL is already configured.

**Review by atodorov — CHANGES_REQUESTED** [review]

File: tcms/core/tests/test_middleware.py [hunk 5]
```diff
@@ -20,3 +21,25 @@ def test_unapplied_migrations(self):
...
+class TestCheckSettingsMiddleware(test.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.base_url_error_message = (
+            'Base URL is not configured! See '
+            '<a href="https://kiwitcms.readthedocs.io/en/latest/admin.html'
+            '#configure-kiwi-s-base-url">documentation</a> and <a href="'
+            '/admin/sites/site/1/change/">change it</a>!'
+        )
```
Comment (atodorov) [inline tcms/core/tests/test_middleware.py:35]: String should be translated.

**Review by atodorov — CHANGES_REQUESTED** [review]
Format with black or see CI complaints.

File: tcms/core/tests/test_middleware.py [hunk 6]
```diff
@@ -12,11 +14,33 @@
...
+
+class TestCheckSettingsMiddleware(test.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.base_url_error_message = _(
+            "Base URL is not configured! See "
+            '<a href="https://kiwitcms.readthedocs.io/en/latest/admin.html'
+            '#configure-kiwi-s-base-url">documentation</a> and <a href="'
+            '/admin/sites/site/1/change/">change it</a>!'
```
Comment (atodorov) [inline tcms/core/tests/test_middleware.py:35]: String format should be exactly the same as in middleware.py in order for the translation to work.

**Review by atodorov — CHANGES_REQUESTED** [review]

**Pull request merged**